### DISCUSSION TITLE
Electric Heat Power sensor not providing the same data as on App

### DISCUSSION THREAD

**Discussion by resain** [tomasmcguinness/homeassistant-mixergy#17] [Q&A] [answered]
Hi,
Just had a Mixergy tank installed and have installed the HA integration with no issues, however comparing the data from the App/website I've noticed a discrepancy between the HA Electric Heat sensor and the app. For example I have a single electrical heating event which occurs at 3:00am (Cheap Rate) for approx 30mins see below.

![mixergy1](https://github.com/tomasmcguinness/homeassistant-mixergy/assets/17032920/eba4eaae-113d-4275-943d-b6fcad0b35e5)

The HA sensor however shows full power for 2 Hours! see below

![mixergy2](https://github.com/tomasmcguinness/homeassistant-mixergy/assets/17032920/5d97c6d8-8e05-4075-9cbe-b1141859bec0)

The electrical power is configured as a custom heat source, which ends at 5:00am which may be relevant.

Is this expected behaviour?

Any advice would be much appreciated as I'm keen to get accurate data into my HA energy dashboard.

Many Thanks in advance!!

**Comment by resain** [comment]
Ok so had a look at the code, so can answer my own question, this is unfortunately expected as the power sensor is generated from the electric heat sensor state (on or off) which does not appear to reflect the actual heating time!

Does anyone know where to look for the API docs? 

The raw energy data is defiantly in there somewhere

**Reply by tomasmcguinness** [reply]
Ohhhhh, perhaps there is a conflict with the scheduled to be on vs the element being actually on?? My integration must be using the pink vs green on the graph. I will take a look! I had no idea it was wrong, so thanks for flagging this!!! I’ll check my own data too. It would mean the same step every day, so this should be obvious.

**Reply by resain** [reply]
So I have done some more research and appear to have some weirdness where the heating event stops unexpectedly for some reason but that doesn't get reported correctly in the state, anyway I am pursuing that with Mixergy. I did run up your test program and looking at the returned results I can see that there are values for voltage and current which I assume are for the immersion? if so and these could be exposed as entities that that would give the possibility of generating accurate power and energy values (using power calc or similar). I can have a look at the code but not got any experience of how HA integrations work so may need some guidance! Or shall I just add a feature request?

**Reply by tomasmcguinness** [reply]
I could certainly expose it, but it’s very strange that it’s not reporting as stopped. I’ve not had an opportunity to look at it in more detail yet as I’m holiday. Does it consistently match the schedule? That can’t be a coincidence.

**Reply by resain** [reply, marked as answer]
So a quick update, the problem was an incorrectly set thermostat which has now been corrected so I don't get any temperature trips and the pink and green traces coincide, obviously the temperature trip is not monitored by the Mixergy controller! My immersion heater only take 2.6kW when heating so there is still a discrepancy in the power values generated by the integration but I've worked around this using powercalc for the time being, but I will have another look at the code when I have more time to see if a more generic approach can be implemented. Many thanks
